corehq/apps/data_dictionary/models.py

```diff
@@ -71,6 +97,15 @@ class DataType(models.TextChoices):
     )
     group = models.TextField(default='', blank=True)
     index = models.IntegerField(default=0, blank=True)
+    group_id = models.ForeignKey(
```

Comment:
I think this should be named `group` (Django automatically adds the `_id` suffix on the db column). Then `db_column="group_id"` can be removed below, and the related group object can be referenced by `case_property.group`.

Reply:
Ah, I missed the `group` text column. However, I think this name will be confusing because  `case_property.group_id` will return a `CasePropertyGroup` object (not an id, which is typically an integer or a string).

Reply:
Yeah, the name can be a little confusing, i can make a comment regarding what `group_id` returns, that can help in reducing this confusion.

fyi @mkangia

Reply:
Hey @millerdev 

@pxwxnvermx and I had some back and forth on the approach to be taken here.

We were initially considering renaming of the old column and adding the new one like you suggested, but that seemed brittle considering that it could lead to issues on the environment in between the time migrations are run and the code update is done for webworkers.
We could also, do the renaming first in a deploy separately but doing any sort of renaming could cause issues because of the reason shared above.
So we decided to go with this approach above.

We would remove the "group" column eventually.
I believe then, we could just rename this `group_id` to group in code in model, which would need no migration since the column name would remain the same (I believe @pxwxnvermx confirmed this locally), and also update any code references from group_id to group on objects.

Open to suggestions/recommendations here.

Reply:
I think it's fine to use `group_id` as a transitional name as long as it gets renamed to `group` before this effort is finished. Maybe add a comment to indicate the reason and plan for eventual state?

Reply:
Thinking about this more, maybe consider using `group_obj` rather than `group_id` to convey that the property returns an object, not an id?

Reply:
Sounds good to me, I leave the final call to @pxwxnvermx

Reply:
Yeah, this looks good to me too.

Reply:
Fixed. [8aa1d2e](https://github.com/dimagi/commcare-hq/pull/32799/commits/8aa1d2e1ebff9ab9f462e49e62166fd5e64d75d9)